### PR TITLE
[SPARK-36272][SQL][TEST] Change shuffled hash join metrics test to check relative value of build size

### PR DESCRIPTION
**Pull request by c21**
### What changes were proposed in this pull request?

This is a follow up of https://github.com/apache/spark/pull/33447, where the unit test is disabled, due to failure after memory setting changed. I found the root cause is after https://github.com/apache/spark/pull/33447, in unit test, Spark memory page byte size is changed from `67108864` to `33554432` [1]. So the shuffled hash join build size is also changed accordingly due to [memory page byte size change](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala#L457). Previously the unit test is checking the exact value of build size, so it no longer works. Here we change the unit test to verify the relative value of build size, and it should work.

[1]: I printed out the memory page byte size explicitly in unit test - `org.apache.spark.SparkException: chengsu pageSizeBytes: 33554432!` in https://github.com/c21/spark/runs/3186680616?check_suite_focus=true .

### Why are the changes needed?

Make previously disabled unit test work.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Changed unit test itself.

### CODE REVIEW

**Comment by c21** [comment]
Will check test result tomorrow to keep investigation.

**Comment by c21** [comment]
After checking the test result - https://github.com/c21/spark/runs/3146847704?check_suite_focus=true, it looks like the BitSet/OpenHashSet data size (used for full outer join) is not changed, but the `HashedRelation` size itself reduces by half. Still checking.

**Comment by c21** [comment]
I found the root cause is after https://github.com/apache/spark/pull/33447, in unit test, Spark memory page byte size is changed from `67108864` to `33554432` (https://github.com/c21/spark/runs/3186680616?check_suite_focus=true). So the shuffled hash join build size is also changed accordingly due to [memory page byte size change](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala#L457). Previously the unit test is checking the exact value of build size, so it no longer works. Here we change the unit test to verify the relative value of build size, and it should work.

**Review by HyukjinKwon — APPROVED** [review]
LGTM, thanks @c21.

**Comment by HyukjinKwon** [comment]
Merged to master, branch-3.2 and branch-3.1.

**Comment by c21** [comment]
Thank you @HyukjinKwon for review!

File: sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala [hunk 1]
```diff
@@ -396,27 +396,25 @@ class SQLMetricsSuite extends SharedSparkSession with SQLMetricsTestUtils
...
         case s: ShuffledHashJoinExec => s
       }
       assert(fojPlan.isDefined, "The query plan should have shuffled hash join")
-      testMetricsInSparkPlanOperator(fojPlan.get,
-        Map("numOutputRows" -> fojRows, "buildDataSize" -> fojBuildSize))
+      testMetricsInSparkPlanOperator(fojPlan.get, Map("numOutputRows" -> fojRows))
+      val fojBuildSize = fojPlan.get.metrics("buildDataSize").value
 
       // Test right outer join as well to verify build data size to be different
       // from full outer join. This makes sure we take extra BitSet/OpenHashSet
```
Comment (viirya) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala:420]: From the comment, looks like that it originally only wants to verify two build data sizes are different.

Comment (c21) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala:420]: well it's actually better to verify build size of full outer is larger than build size of right outer. I can update comment if you think it's needed.

Comment (viirya) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala:420]: That's fine. The current change looks good.

**Pull request closed without merging**